File: android/app/src/main/java/com/hsklearn/app/data/model/ReadingPassage.kt
package com.hsklearn.app.data.model

import kotlinx.serialization.Serializable

@Serializable
data class ReadingPassage(
    val id: Int,
    val topic: String,
    val title: String,
    val titlePinyin: String,
    val passage: String,
    val passagePinyin: String,
    val targetWords: List<String>,
    val questions: List<ComprehensionQuestion>,
)

@Serializable
data class ComprehensionQuestion(
    val question: String,
    val options: List<String>,
    val correctIndex: Int,
    val explanation: String,
)

File: backend/HskLearn.Api/Models/ReadingPassage.cs
namespace HskLearn.Api.Models;

public record ReadingPassage(
    int Id,
    string Topic,
    string Title,
    string TitlePinyin,
    string Passage,
    string PassagePinyin,
    List<string> TargetWords,
    List<ComprehensionQuestion> Questions
);

public record ComprehensionQuestion(
    string Question,
    List<string> Options,
    int CorrectIndex,
    string Explanation
);
